CMakeLists.txt

```diff
@@ -1244,9 +1003,224 @@ if(CONFIG_USERSPACE)
   )
 endif()
 
+if(CONFIG_USERSPACE OR CONFIG_HAS_DTS)
+  list(APPEND LINKER_PASS_${ZEPHYR_CURRENT_LINKER_PASS}_DEFINE "DEVICE_HANDLES_PASS1")
```

Comment:
Why are you appending `DEVICE_HANDLES_PASS1` even if DTS is disabled?

Reply:
nice catch.

That would probably have been something I would notice if I was reviewer :laughing: 

Fixed, moved together with gen_handles code where it rightfully belongs.